yucca/preprocessing/YuccaPreprocessor.py

```diff
@@ -675,6 +695,21 @@ def sanity_check_niftis(self, images, label, subject_id):
                     f"One is: {get_nib_orientation(images[0])} while another is {get_nib_orientation(image)}"
                 )
 
+    def cast_to_numpy_array(self, images: list, label=None, classification=False):
+        dtype = "float32" if not self.allow_missing_modalities else "object"
```

Comment:
So the reason for using object arrays is that you can not do what you are suggesting with arrays of uneven sizes
such as:
`x = np.array([[3.2, 4.2], [np.nan]], dtype=np.float32)`

So if we were to use np.nan we would need to fill an entire array the same size as the other modalities. And this was the thing we wanted to avoid as that would take up all the space we're trying to save.

The object array is the only np array that allows combining arrays of different sizes. That is also why we use it for classification to combine e.g. an image of shape [224, 224] with a label of shape [1]

Reply:
So to answer your questions more methodically:
1. For classification, yes. The others no if we want to allow arrays of uneven sizes as they may be object arrays in that case and float if uneven sizes are not allowed
2. Yes, good idea.
3. also good idea.
4. Was what I responded to in the first comment more or less

Reply:
Aha, makes sense with arrays of different length :-) I am still a bit confused though. As far as the code is written now, the dtype perfectly aligns with the below branches, such that the following will always be true:

```python

if label is None:  # self-supervised
            images = np.array(images, dtype=np.float32)  # <--- always float
        elif classification:
            images.append(label)
            images = np.array(images, dtype=np.float32)  # <--- always float
        elif self.allow_missing_modalities:  # Segmentation with missing modalities
            images.append(np.array(label)[np.newaxis])
            images = np.array(images, dtype=np.object)      # <--- always object
        else:  # Standard segmentation
            images = np.vstack((np.array(images), np.array(label)[np.newaxis]))  # <--- always float (if it was provided)
        return images
```